src/votekit/election_types.py

```diff
@@ -1,98 +1,125 @@
-
-        # if number of remaining candidates equals number of remaining seats
-        if len(candidates) == self.seats - len(self.elected):
-            # TODO: sort remaing candidates by vote share
-            self.elected.update(set(candidates))
-            return profile, Outcome(
-                elected=self.elected,
-                eliminated=self.eliminated,
-                remaining=set(candidates),
-                votes=fp_votes,
-            )
-
-        for candidate in candidates:
-            if fp_votes[candidate] >= self.threshold:
-                self.elected.add(candidate)
-                candidates.remove(candidate)
-                ballots = self.transfer(candidate, ballots, fp_votes, self.threshold)
-
-        if self.next_round():
+        ##TODO:must change the way we pass winner_votes
+        remaining: list = self.election_state.remaining
+        ballots: list = self.election_state.profile.get_ballots()
+        fp_votes: dict = compute_votes(remaining, ballots)
+        fp_order = [
+            y[0] for y in sorted(fp_votes.items(), key=lambda x: x[1], reverse=True)
+        ]
+        elected = []
+        eliminated = []
+
+        # if number of remaining candidates equals number of remaining seats, everyone is elected
+        if len(remaining) == self.seats - len(self.election_state.get_all_winners()):
+            elected = fp_order
+            remaining = []
+            ballots = []
+            # TODO: sort remaining candidates by vote share
+
+        # elect all candidates who crossed threshold
+        elif fp_votes[fp_order[0]] >= self.threshold:
+            for candidate in fp_order:
+                if fp_votes[candidate] >= self.threshold:
+                    elected.append(candidate)
+                    remaining.remove(candidate)
+                    ballots = self.transfer(
+                        candidate, ballots, fp_votes, self.threshold
+                    )
+        # since no one has crossed threshold, eliminate one of the people
+        # with least first place votes
+        else:
             lp_votes = min(fp_votes.values())
             lp_candidates = [
-                candidate for candidate, votes in fp_votes.items() if votes == lp_votes
+                candidate for candidate in fp_order if fp_votes[candidate] == lp_votes
             ]
             # is this how to break ties, can be different based on locality
-            lp_cand = random.choice(lp_candidates)
-            ballots = remove_cand(lp_cand, ballots)
-            candidates.remove(lp_cand)
-            self.eliminated.add(lp_cand)
-
-        return PreferenceProfile(ballots=ballots), Outcome(
-            elected=self.elected,
-            eliminated=self.eliminated,
-            remaining=set(candidates),
-            votes=fp_votes,
+            eliminated.append(random.choice(lp_candidates))
+            ballots = remove_cand(eliminated[0], ballots)
+            remaining.remove(eliminated[0])
```

Comment:
we should set `lp_cand = random.choice(lp_candidates)` as a variable so a) its clearer what is happening vs just indexing the list and b) I'm not sure thats the only way to break ties in this instance so it leaves room to abstract that

Reply:
Done